### ISSUE TITLE
Unable to mock ContactsEndPoint's Where() method

### ISSUE THREAD

**Issue by Alonnj** [XeroAPI/Xero-Net#369]
Greetings,
As part of unit tests, I would like to mock the where() method in ContactsEndPoint class.
However, since ContactsEndPoint's constructor is internal, I'm:
1. Unable to make a stub of this class (using your recommendation - Rhino.Mocks)
2. Unable to create a class which will inherit from ContactsEndPoint.

How can I bypass this problem? please help.
Thanks,
Alon.

**Comment by Alonnj** [comment]
Yes, mocking the interface works fine, but the interface has a property called "Contacts" whose type is ContactsEndPoint.
And in ContactsEndPoint, the where() method returns ContactsEndPoint.
In my mocked interface, I need "Contacts" to returns a mocking of ContactsEndPoint, so the where() method in this mocked class will return the same type.
But I can't create such as class, as I've explained above.

**Comment by MatthewSteeples** [comment]
I've just noticed why we don't suffer from this problem. We've ended up making a fork of this library (https://github.com/matthewsteeples/Xero-Net/) and we've fixed that in there. We do have a PR with _all_ of our changes in it, but some of them are breaking (changing enum numbers) so it's not likely to get merged in.

If you want to create a PR for just this change then the commit you're after is https://github.com/XeroAPI/Xero-Net/pull/323/commits/9e943d0cda7533d3e2743bb7ef13e25ca847c773 but I'm afraid it's not something we'll be able to get round to for a while
